File: Lab/Lab5/lab5.java
public class lab5{
    public boolean isValidBST(TreeNode root) {
        return validBST(root, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY);  
    }
    private boolean validBST(TreeNode node, double min, double max){
        if(node == null){
            return true;
        }
        if(!(node.val > min && node.val < max)){
              return false;
        }
        return (validBST(node.left, min, node.val) && validBST(node.right, node.val, max));
    }
}

File: Lab/Lab5/lab5.py
class lab5:
   def isValidBST(self, root):
        def validBst(node, min, max):
            if node == None:
                return True
            
            if not (node.val > min and node.val < max):
                return False

            return (validBst(node.left, min, node.val) and validBst(node.right, node.val, max))

        return validBst(root, float("-inf"), float("inf"))
